### PR TITLE
[#1113][#1224] improvement(UI): TS support and improve global request handing

### PR DESCRIPTION
**Pull request by ch3yne**
### What changes were proposed in this pull request?
 
1. Support TypeScript
2. Improve global reauest handing and add error messages

### Why are the changes needed?

Fix: #1113 
Fix: #1224 
Fix: #1225 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by justinmclean** [comment]
There is no need to add web/licenses/typescript.txt as we already have the ALv2 text in full.

**Comment by justinmclean** [comment]
You also need to add the product that is "MIT License Copyright (c) 2020-present, Vben" to LICENSE.

File: web/LICENSE.bin [hunk 1]
```diff
@@ -203,15 +203,20 @@
...
    Apache Zeppelin
    ./web/WEB-INF/web.xml
 
    @swc/helpers
+   typescript
 
    This product bundles various third-party components also under the
    MIT license.
 
+   MIT License Copyright (c) 2020-present, Vben
```
Comment (ch3yne) [inline web/LICENSE.bin:216]: @justinmclean Is this the correct format for adding it?

**Comment by ch3yne** [comment]
> There is no need to add web/licenses/typescript.txt as we already have the ALv2 text in full.

Removed.

File: web/LICENSE.bin (hunk 1 above)
Comment (justinmclean) [reply web/LICENSE.bin:216]: you need to list the product name not the license and copyright

Comment (ch3yne) [reply web/LICENSE.bin]: Retained only Vben

**Comment by qqqttt123** [comment]
Does this pull request need to back port branch 0.3?

**Comment by ch3yne** [comment]
> Does this pull request need to back port branch 0.3?

I think it's needed.

File: web/lib/enums/httpEnum.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+export enum ContentTypeEnum {
```
Comment (jerryshao) [inline web/lib/enums/httpEnum.ts:25]: Is this file copied from other projects?

Comment (jerryshao) [reply web/lib/enums/httpEnum.ts:25]: It is better to add a line like "Referred from core/src/main/java/org/apache/iceberg/rest/RESTClient.java" on the top of the code.

All the copied code need this line.

image

Comment (justinmclean) [reply web/lib/enums/httpEnum.ts:25]: You also need to update LICENSE and LICENSE.bin

Comment (justinmclean) [reply web/lib/enums/httpEnum.ts:25]: Also if it was copied from Iceberg what does the file have an MIT license header?

Comment (jerryshao) [reply web/lib/enums/httpEnum.ts:25]: The pasted image is just an example.

File: web/lib/utils/axios/checkStatus.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    case 504:
+      errMessage = 'Network timeout!'
+      break
+
+    case 505:
+      errMessage = 'The http version does not support the request!'
+      break
+
+    default:
```
Comment (jerryshao) [inline web/lib/utils/axios/checkStatus.ts:95]: Are these error messages all defined messages? I think we (Gravitino) have our error message, should we use the Gravitino's returned error message?

**Review by jerryshao — DISMISSED** [review]
LGTM.

**Comment by jerryshao** [comment]
Seems there're some building issues we need to fix. @ch3yne

**Comment by jerryshao** [comment]
@justinmclean can you please help to check Iicense related things, thanks!

**Review by jerryshao — APPROVED** [review]

**Pull request merged**